src/archive/src/main.rs

```diff
@@ -0,0 +1,394 @@
+use candid::{CandidType, Deserialize, Principal};
```

Comment:
Would be nice to have a few words about the memory layout, I'm struggling to wrap my head around it